src/Compilers/Core/Portable/Compilation/Expression.cs

```diff
@@ -212,26 +220,26 @@ public void Accept(OperationVisitor visitor)
 
         private sealed class CompoundAssignmentExpression : ICompoundAssignmentExpression
         {
-            public CompoundAssignmentExpression(IReferenceExpression target, IExpression value, BinaryOperationKind binaryKind, IMethodSymbol operatorMethod, SyntaxNode syntax)
+            public CompoundAssignmentExpression(IReferenceExpression target, IOperation value, BinaryOperationKind binaryKind, IMethodSymbol operatorMethod, SyntaxNode syntax)
             {
                 this.Target = target;
                 this.Value = value;
-                this.BinaryKind = binaryKind;
-                this.Operator = operatorMethod;
+                this.BinaryOperationKind = binaryKind;
```

Comment:
I'd make the parameter name match the new property name: binaryKind -> binaryOperationKind.